pkg/wsclient/wsclient.go

```diff
@@ -204,25 +204,19 @@ func (w *wsClient) readLoop() {
 	l := log.L(w.ctx)
 	for {
 		mt, message, err := w.wsconn.ReadMessage()
-
-		// Check there's not a pending send message we need to return
-		// before returning any error (do not block)
-		select {
-		case <-w.sendDone:
-			l.Debugf("WS %s closing reader after send error", w.url)
-			return
-		default:
-		}
-
-		// return any error
 		if err != nil {
 			l.Errorf("WS %s closed: %s", w.url, err)
 			return
 		}
 
 		// Pass the message to the consumer
 		l.Tracef("WS %s read (mt=%d): %s", w.url, mt, message)
-		w.receive <- message
```

Comment:
Previous code could wait indefinitely here, if receiver was trying to perform a send that failed.